src/lib.rs

```diff
@@ -448,6 +479,15 @@ impl Exporter {
         Builder::new_listener(listener)
     }
 
+    /// TODO
+    #[must_use = "not using the guard will result in the exporter returning the prometheus data \
+                  immediately over http"]
+    pub fn wait(&self) -> MutexGuard<'_, ()> {
```

Comment:
The name is very counterintuitive: when one calls that method, their goal is not to wait for some event, but to lock the exporter while they're performing some operations. Wouldn't it be better to call it `lock`?